services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -251,6 +244,69 @@ async def list_paths(
 
         return paths_metadata, next_cursor, total
 
+    async def compute_path_total_size(self, user_id: UserID, *, path: Path) -> ByteSize:
+        """returns the total size of an arbitrary path"""
+        # check access rights first
+        project_id = None
+        with contextlib.suppress(ValueError):
+            # NOTE: we currently do not support anything else than project_id/node_id/file_path here, sorry chap
+            project_id = ProjectID(path.parts[0])
+        async with self.engine.connect() as conn:
+            accessible_projects_ids = await get_accessible_project_ids(
+                conn, user_id=user_id, project_id=project_id
+            )
+
+        # use-cases:
+        # 1. path is not a valid StorageFileID (e.g. a project or project/node) --> all entries are in the DB (files and folder)
+        #   2. path is valid StorageFileID and not in the DB --> entries are only in S3
+        #   3. path is valid StorageFileID and in the DB --> return directly from the DB
+
+        use_db_data = True
+        with contextlib.suppress(ValidationError):
+            file_id: StorageFileID = TypeAdapter(StorageFileID).validate_python(
+                f"{path}"
+            )
+            # path is a valid StorageFileID
+            async with self.engine.connect() as conn:
```

Comment:
storage still does not separate repository, right?

Reply:
well it does mostly. I still do not use the transaction since you said it has issues on Monday.
you might see that only the connection is acquired here. so all the calls to the database are actually wrapped. and that since a very long time.

Reply:
Here I meant having something like a `repository class that handles all these connections etc

Similar
https://github.com/itisfoundation/osparc-simcore/blob/5732a12e07e63d5ce55010ede9b9ab543bb9b278/packages/postgres-database/src/simcore_postgres_database/utils_projects.py#L21-L41


Now, regarding `transaction_context`, i am more confident that this works fine on all the way we have used so far. What I am not sure is whether we are missing some edge cases. Remember that the commit policy can be configured as well upstream (e.g. at the engine or at the connection level). I believe these cases are still not fully covered. Now, if you do not have those, you should definitively uses these the helpesr in `utils_repos.py`

Reply:
will add the transaction thing in another PR: https://github.com/ITISFoundation/osparc-simcore/issues/7312